plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/TrinoCatalog.java

```diff
@@ -71,6 +71,8 @@ Transaction newCreateTableTransaction(
             String location,
             Map<String, String> properties);
 
+    void registerTable(ConnectorSession session, SchemaTableName tableName, String tableLocation, Optional<String> metadataFileName);
```

Comment:
FYI @danielcweeks you'll need to add an implementation to your REST catalog PR if this merges first. https://github.com/trinodb/trino/pull/13294